apache-rat-core/src/main/java/org/apache/rat/inspector/AbstractInspector.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ */ 
+package org.apache.rat.inspector;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.rat.license.ILicense;
+
+public abstract class AbstractInspector implements Inspector {
+    private final Type type;
+    private final String name;
+
+    public AbstractInspector(Type type, String name) {
+        this.type = type;
+        this.name = name;
+    }
+
+    @Override
+    public Type getType() {
+        return type;
+    }
+
+    @Override
+    public String getCommonName() {
+        return name;
+    }
+
+    public static Inspector license(ILicense license, Inspector matcher) {
+        return new AbstractInspector(Inspector.Type.License, "license") {
```

Comment:
Would one want to extract all of these license-config-related constants into a class where we could easily add documentation about the semantics of each option/value/parameter?